data-prepper-plugins/csv-processor/src/main/java/com/amazon/dataprepper/plugins/processor/csv/CSVProcessor.java

```diff
@@ -12,22 +12,60 @@
 import com.amazon.dataprepper.model.processor.AbstractProcessor;
 import com.amazon.dataprepper.model.processor.Processor;
 import com.amazon.dataprepper.model.record.Record;
+import com.fasterxml.jackson.databind.MappingIterator;
+import com.fasterxml.jackson.dataformat.csv.CsvMapper;
+import com.fasterxml.jackson.dataformat.csv.CsvParser;
+import com.fasterxml.jackson.dataformat.csv.CsvSchema;
+
+import java.io.IOException;
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.List;
 
 /**
  * Processor to parse CSV data in Events.
  *
  */
 @DataPrepperPlugin(name="csv", pluginType = Processor.class, pluginConfigurationType = CSVProcessorConfig.class)
 public class CSVProcessor extends AbstractProcessor<Record<Event>, Record<Event>> {
+    private final CSVProcessorConfig config;
+
     @DataPrepperPluginConstructor
     public CSVProcessor(final PluginMetrics pluginMetrics, final CSVProcessorConfig config) {
         super(pluginMetrics);
-        throw new UnsupportedOperationException();
+        this.config = config;
     }
 
     @Override
     public Collection<Record<Event>> doExecute(final Collection<Record<Event>> records) {
+        final CsvMapper mapper = createCsvMapper();
+        final CsvSchema schema = createCsvSchema();
+
+        for (final Record<Event> record : records) {
+
+            final Event event = record.getData();
+
+            final String message = event.get(config.getSource(), String.class);
+            final boolean userDidSpecifyHeaderEventKey = (config.getColumnNamesSourceKey() != null);
+            final boolean thisEventHasHeaderSource = userDidSpecifyHeaderEventKey && event.containsKey(config.getColumnNamesSourceKey());
+
+            try {
+                final MappingIterator<List<String>> messageIterator = mapper.readerFor(List.class).with(schema).readValues(message);
+
+                // otherwise the message is empty
+                if (messageIterator.hasNextValue()) {
+                    final List<String> row = messageIterator.nextValue();
+                    final List<String> header = parseHeader(event, thisEventHasHeaderSource, mapper, schema);
+                    putDataInEvent(event, header, row);
+                }
+
+                if (thisEventHasHeaderSource && config.isDeleteHeader()) {
+                    event.delete(config.getColumnNamesSourceKey());
+                }
+            } catch (IOException e) {
+                System.out.println(e.toString());
```

Comment:
I think this should be an `error` log instead of printing to stdout. Here are some references on that

Instantiate logger: https://github.com/opensearch-project/data-prepper/blob/0a5910b789e004a0b69b8c053660db8b3a2da835/data-prepper-plugins/grok-prepper/src/main/java/com/amazon/dataprepper/plugins/prepper/grok/GrokPrepper.java#L58

Write error log: https://github.com/opensearch-project/data-prepper/blob/0a5910b789e004a0b69b8c053660db8b3a2da835/data-prepper-plugins/grok-prepper/src/main/java/com/amazon/dataprepper/plugins/prepper/grok/GrokPrepper.java#L124